### ISSUE TITLE
How to add UNknown label for untrained images

### ISSUE THREAD

**Issue by guruvishnuvardan** [AISangam/Facenet-Real-time-face-recognition-using-deep-learning-Tensorflow#4]
HI,

I am using this code to identify unknown faces that are not trained, so Can you please let me know, where to add "Unknown" label, so that I can take screen shot of those unknown persons to train later.

Thanks
Guru

**Comment by albertoZurini** [comment]
Try using this other library https://github.com/ageitgey/face_recognition

**Comment by SijinJohn** [comment]
> Try using this other library https://github.com/ageitgey/face_recognition

Use what library??i didn't understand.

**Comment by khan958** [comment]
@albertoZurini @SijinJohn  
Same problem with me? Have you train this model for unknown person?

**Comment by albertoZurini** [comment]
What is exactly your problem? The readme is enough easy to understand. If you could be more specific I'd be glad to help you

**Comment by SijinJohn** [comment]
@albertoZurini the model recognizes an unknown face as member from the dataset. How to solve this issue?

**Comment by albertoZurini** [comment]
Okay, now I understood what you meant (it's been a while since i last used these libraries). Before you start using these libraries I highly recommend you to read the paper of the original research (it should be found on Arxiv).
So what does the neural network do? You basically input it to images and it will output the difference between the faces as a float. So in order to check if a face is the actual face of a person you threshold the output of the net (eg. if out>2 that isn't the faces of the person). That's the approach I used in my project

**Comment by SijinJohn** [comment]
if i have trained the model for faces A,B,C and if D comes during testing, it may predict D as A,B,C even after reducing the threshold value.

**Comment by albertoZurini** [comment]
Yeah, that's a limitation of this library. Try using the other one which I've linked

**Comment by khan958** [comment]
@albertoZurini and the other thing is that ? How can i export my result from this experiment?

**Comment by khan958** [comment]
i want to export my result from this experiment. So can you help me how to
generate result from this experiments .

**Comment by khan958** [comment]
when i run python identify_face_image.py  it shows me this ?
can you tell me please what is this ?
result indices :7 sometime 5
 [[0.00921852 0.07074382 0.00716605 0.00421573 0.02599692 0.01865331
  0.01690321 0.01006578 0.06874712 0.00450729 0.01254694 0.02206854
  0.02750619 0.00432872 0.03058342 0.00813806 0.64345926 0.00723246
  0.00791865]]
[0.64345926]

**Comment by albertoZurini** [comment]
> @albertoZurini and the other thing is that ? How can i export my result from this experiment?

I'm pretty sure you can't export the data from this neural network to the other due to architecture differences

**Comment by khan958** [comment]
@albertoZurini  and what about this one ?

https://github.com/davidsandberg/facenet

**Comment by albertoZurini** [comment]
> @albertoZurini and what about this one ?
> 
> https://github.com/davidsandberg/facenet

I've tried to get it running, but with no result. If you can get it that would be the best NN for facial recognition.
